### PR TITLE
10  generic invite

### LINKED ISSUE #10: implement the generic invite dialog in the footer

### PR DESCRIPTION
**Pull request by dkuntze**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #10 

Test URLs:
- Before: https://main--basf-retailfirst--hlxsites.hlx.page/
- After: https://10--generic-invite--basf-retailfirst--hlxsites.hlx.page/

### CODE REVIEW

File: scripts/scripts.js [hunk 1]
```diff
@@ -121,6 +121,78 @@ function decorateVideos(el) {
...
+    hideModal(event);
+  });
+  modal.querySelector('.row > input[type=button]').addEventListener('click', submitInvite);
+  modal.querySelector('div > .close-invite').addEventListener('click', (event) => {
+    hideModal(event);
+  });
+}
+
+function makeInviteModal() {
+  const modal = document.querySelector("[title='Share with a friend or colleague.']");
```
Comment (ramboz) [inline scripts/scripts.js]: Feels like a fragile selector if the content is coming from the document. I'd suggest making a link with a hashtag target like `#invite` instead and select on that

File: scripts/scripts.js [hunk 2]
```diff
@@ -121,6 +121,78 @@ function decorateVideos(el) {
...
+  if (modal) {
+    modal.addEventListener('click', () => {
+      showModal();
+    });
+    // eslint-disable-next-line no-script-url
+    modal.href = 'javascript:void(0);';
```
Comment (ramboz) [inline scripts/scripts.js]: Any particular reason to `void` the `href` here? You could do `ev.preventDefault()` in the handler to forcibly disable the default behavior instead

File: scripts/scripts.js [hunk 3]
```diff
@@ -121,6 +121,78 @@ function decorateVideos(el) {
...
+  const name = event.target.parentNode.parentNode.querySelector('.row > input[id=name]').value;
+  const email = event.target.parentNode.parentNode.querySelector('.row > input[id=email]').value;
+  const validRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+
+  if (name === '') {
+    messageDiv.innerText = 'Name is required';
+    valid += 1;
+  }
+  if (!email.match(validRegex)) {
+    messageDiv.innerText = 'Valid email is required';
```
Comment (ravkiran) [inline scripts/scripts.js:159]: could we style the message div to show errors in red color, like in their current site?

Comment (dkuntze) [reply scripts/scripts.js:159]: That's fair. I think the success message is in green too.

File: scripts/scripts.js (hunk 1 above)
Comment (dkuntze) [reply scripts/scripts.js]: The content isn't coming from a document.

File: scripts/scripts.js (hunk 2 above)
Comment (dkuntze) [reply scripts/scripts.js]: No reason. Going old school. I'll change it.

File: scripts/scripts.js (hunk 1 above)
Comment (dkuntze) [reply scripts/scripts.js]: nevermind, I was thinking of something else. It is content - and I changed it in footer (doc) and the code.

**Review by ramboz — APPROVED** [review]

**Pull request merged**